corehq/form_processor/document_stores.py

```diff
@@ -60,10 +60,17 @@ def __init__(self, domain, case_type=None):
         self.domain = domain
         self.case_type = case_type
 
-    def get_document(self, doc_id):
+    def get_document(self, doc_id, *, external_id=None):
```

Comment:
Just sharing that this was the primary reason for why I considered a new class in the draft PR.

This changes the signature of the method from the base class which I am not comfortable with and I feel makes the code brittle. It's more of a personal thing though I don't know what others think about it.